Core.Collectors/Web/HttpResponseSignature.cs

```diff
@@ -1,29 +1,39 @@
 ﻿// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
+using Microsoft.CloudMine.Core.Collectors.Collector;
 using Newtonsoft.Json.Linq;
 using System;
+using System.Collections.Generic;
 using System.Net;
 using System.Text.RegularExpressions;
 
 namespace Microsoft.CloudMine.Core.Collectors.Web
 {
-    public class HttpResponseSignature
+    public class HttpResponseSignature : IAllowListStatus
     {
         private readonly HttpStatusCode statusCode;
         private readonly Regex responseMessageRegex;
         private readonly Func<JObject, bool> matcher;
+        private readonly List<CollectionNode> continuation;
```

Comment:
Why don't we also make this a function-based continuation (just in case we need recursion here as well)? That would also keep things similar.